foundationdb/src/options.rs

```diff
@@ -0,0 +1,615 @@
+    SetVersionstampedKey(Vec<u8>),
+    /// value to versionstamp and set
+    ///
+    /// Transforms ``param`` using a versionstamp for the transaction. Sets ``key`` in the database to the transformed parameter. A versionstamp is a 10 byte, unique, monotonically (but not sequentially) increasing value for each committed transaction. The first 8 bytes are the committed version of the database. The last 2 bytes are monotonic in the serialization order for transactions. WARNING: At this time versionstamped values are not compatible with the Tuple layer.
+    SetVersionstampedValue(Vec<u8>),
+    /// value to check against database value
+    ///
+    /// Performs lexicographic comparison of byte strings. If the existing value in the database is not present, then ``param`` is stored. Otherwise the smaller of the two values is then stored in the database.
+    ByteMin(Vec<u8>),
+    /// value to check against database value
+    ///
+    /// Performs lexicographic comparison of byte strings. If the existing value in the database is not present, then ``param`` is stored. Otherwise the larger of the two values is then stored in the database.
+    ByteMax(Vec<u8>),
+}
+
+impl MutationType {
+    pub fn code(&self) -> fdb::FDBMutationType {
+        match *self {
+            MutationType::Add(ref _v) => 2,
+            MutationType::And(ref _v) => 6,
+            MutationType::BitAnd(ref _v) => 6,
+            MutationType::Or(ref _v) => 7,
+            MutationType::BitOr(ref _v) => 7,
+            MutationType::Xor(ref _v) => 8,
+            MutationType::BitXor(ref _v) => 8,
+            MutationType::Max(ref _v) => 12,
+            MutationType::Min(ref _v) => 13,
+            MutationType::SetVersionstampedKey(ref _v) => 14,
+            MutationType::SetVersionstampedValue(ref _v) => 15,
+            MutationType::ByteMin(ref _v) => 16,
+            MutationType::ByteMax(ref _v) => 17,
+        }
+    }
+}
+
+pub enum ConflictRangeType {
+    /// Used to add a read conflict range
+    Read,
+    /// Used to add a write conflict range
+    Write,
+}
+
+impl ConflictRangeType {
+    pub fn code(&self) -> fdb::FDBConflictRangeType {
+        match *self {
+            ConflictRangeType::Read => 0,
+            ConflictRangeType::Write => 1,
+        }
+    }
+}
+
+pub enum ErrorPredicate {
+    /// Returns ``true`` if the error indicates the operations in the transactions should be retried because of transient error.
+    Retryable,
+    /// Returns ``true`` if the error indicates the transaction may have succeeded, though not in a way the system can verify.
+    MaybeCommitted,
+    /// Returns ``true`` if the error indicates the transaction has not committed, though in a way that can be retried.
+    RetryableNotCommitted,
+}
+
+impl ErrorPredicate {
+    pub fn code(&self) -> fdb::FDBErrorPredicate {
+        match *self {
+            ErrorPredicate::Retryable => 50000,
```

Comment:
Technically these values are `fdb::FDBErrorPredicate_FDB_ERROR_PREDICATE_MAYBE_COMMITTED`, rather than the raw value. Not sure if this is something the generator could write instead?

Reply:
I just tried to change those values to generated one, but there are some mismatches between names on `fdb.options` and names on `fdb_c.h`. Here's a list.

https://gist.github.com/yjh0502/cd19273e15464e5e2c377fc9f3ce3e03

 - Some of them are just singular/plural problem, for example `WATCH/WATCHES`, `BYTE/BYTES`. I can handle it with some brute-force replace rules.
 - It seems that options with `hidden=true` is not available on c header. I filtered those options.
 - There's one remaining non-regular case, `FDBMutationType_FDB_MUTATION_TYPE_SET_VERSIONSTAMPED_KEY`, which should not be converted to `..._KEYS`. I just hard-coded the case :(